src/app/shared/base-components/upload-excel/upload-excel.component.ts

```diff
@@ -0,0 +1,168 @@
+import { DOCUMENT } from '@angular/common';
+import { Component, HostListener, Inject } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { WINDOW } from 'ngx-window-token';
+import * as XLSX from 'xlsx/xlsx.mjs';
+import { ImportValidationService } from 'shared/services/import-validation/import-validation.service';
+import { FieldsConfig } from 'shared/models/admin-import-export.model';
+
+@Component({
+  selector: 'app-import-providers',
+  template: '<div></div>',
+  styleUrls: ['./upload-excel.component.scss']
+})
+export class UploadExcelComponent<T extends { errors: any }, U extends T & { id: number }> {
+  public extendsComponentConfig: FieldsConfig[];
+  public isToggle: boolean;
+  public isLoading: boolean = false;
+  public isWarningVisible: boolean = false;
+  public selectedFile: any = null;
+  public isGoTopBtnVisible: boolean;
+  public columnNamesBase: string[];
+  public standardHeadersBase: string[];
+  public readonly topPosToStartShowing: number = 250;
+
+  public dataSource: U[];
+  public dataSourceInvalid: U[];
+
+  constructor(
+    private readonly importValidationService: ImportValidationService,
+    private readonly translate: TranslateService,
+    @Inject(DOCUMENT) private readonly document: Document,
+    @Inject(WINDOW) private readonly window: Window
+  ) {}
+
+  @HostListener('window:scroll')
+  public checkScroll(): void {
+    const scrollPosition = this.document.documentElement.scrollTop || this.document.body.scrollTop || 0;
+    if (scrollPosition >= this.topPosToStartShowing) {
+      this.isGoTopBtnVisible = true;
+    } else {
+      this.isGoTopBtnVisible = false;
+    }
+  }
+
+  public gotoTop(): void {
+    this.window.scroll({
+      top: 0,
+      left: 0,
+      behavior: 'smooth'
+    });
+  }
+
+  public setColumnNames(columnNames: string[]): void {
+    this.columnNamesBase = columnNames;
+  }
+
+  public setStandardHeaders(headers: string[]): void {
+    this.standardHeadersBase = headers;
+  }
+
+  public resetValues(): void {
+    this.dataSource = null;
+    this.dataSourceInvalid = null;
+    this.isToggle = false;
+    this.isWarningVisible = false;
+  }
+
+  public convertExcelToJSON(file: File): void {
+    const reader: FileReader = new FileReader();
+    reader.onerror = (): void => {
+      alert(this.translate.instant('IMPORT/EXPORT.FILE_READER_WARNING'));
```

Comment:
From a UX perspective, it's better to use dialog boxes.
But since we don't have a universal reusable modal, we can leave it as is and then add it later